Document the GitHub (read-only) integration (#773)
Promptless/docs: src/content/docs/docs/connect/doc-locations/github-repos.mdx

- For GitHub PR triggers, the documentation PR is linked in a comment on the original code PR (you can disable these comments with `suppress_source_pr_comments: true` in your policy)
- For [commit triggers](/docs/connect/triggers/github-commits), you can enable auto-merge to automatically merge documentation PRs as soon as they're created (`auto_merge: true`)

## Read-only integrations (public repositories)

> When a doc collection is backed by a [GitHub (read-only)](/docs/reference/integrations/github-read-only) integration, Promptless publishes through a pull request from a fork into your upstream base branch (public repositories only; private repos need the [standard GitHub App](/docs/reference/integrations/github)). The fork is set up when you configure the doc collection. See the [GitHub (read-only) integration](/docs/reference/integrations/github-read-only) for the canonical reference on how this works.

Reviewer: ```suggestion
When a doc collection is set up for a public repository and backed by a [GitHub (read-only)](/docs/reference/integrations/github-read-only) integration, Promptless publishes through a pull request from a fork into your upstream base branch. (This doesn't work for private repos, which need the [standard GitHub App](/docs/reference/integrations/github).) The fork is set up when you configure the doc collection. See the [GitHub (read-only) integration](/docs/reference/integrations/github-read-only) for how this works.
```

> When a doc collection is set up for a public repository and backed by a [GitHub (read-only)](/docs/reference/integrations/github-read-only) integration, Promptless publishes through a pull request from a fork into your upstream base branch. (This doesn't work for private repos, which need the [standard GitHub App](/docs/reference/integrations/github).) Promptless creates the fork itself — the first time it needs to work in the repository, not when you configure the doc collection. See the [GitHub (read-only) integration](/docs/reference/integrations/github-read-only) for how this works.

Reviewer: ```suggestion
When a doc collection is set up for a public repository and backed by a [GitHub (read-only)](/docs/reference/integrations/github-read-only) integration, Promptless publishes through a pull request from a fork into your upstream base branch. (This doesn't work for private repos, which need the [standard GitHub App](/docs/reference/integrations/github).) Promptless creates the fork itself. See the [GitHub (read-only) integration](/docs/reference/integrations/github-read-only) for how this works.
```
